src/components/Pages/HomePage.js

```diff
@@ -1,20 +1,28 @@
-import React, { useEffect, useState } from 'react';
+import React, { useEffect, useState, useContext } from 'react';
 import { getHighlyRatedEpisodes } from '../../api/api';
 import { TVShowPoster } from '../Main/TvShowPoster';
 import { SeriesSearchForm } from '../SeriesSearch/SeriesSearchForm';
+import { AppContext } from '../../contexts/AppContext';
 import { _ } from '../../utils';
 
 // http://localhost:8080/
 export const HomePage = () => {
+  const { setIsLoading } = useContext(AppContext);
   const [episodesData, setEpisodesData] = useState([]);
 
   const getData = () => {
     const response = getHighlyRatedEpisodes(5);
-    response.then(({ data }) => setEpisodesData(data.results));
+    setIsLoading(true);
+    response.then(({ data }) => {
+      setEpisodesData(data.results);
+      setIsLoading(false);
+    });
   };
 
   useEffect(() => {
     getData();
+
+    return () => {};
```

Comment:
0I am kind of confused by this...Is this just a result of using useEffect cleanup functionality (similar to unmounting?)

Reply:
oh yeah, i left this in. we dont need this return line